### PR TITLE
Adds a storybook

### PR DESCRIPTION
**Pull request by gsimone**
Just a basic storybook with a few of the components, will work on adding all the components

Preview here:  https://drei-storybook-beta.netlify.app

I tried to add an addon for types/propTypes but I'm not really satisfied, suggestions welcome.

# Progress

# Cameras
- [x] PerspectiveCamera
- [x] OrthographicCamera
# Controls
- [x] OrbitControls
- [x] MapControls
- [x] TrackballControls
- [x] TransformControls
# Shapes
- [x] Plane 
- [x] Box 
- [x] Sphere 
- [x] Circle 
- [x] Cone 
- [x] Cylinder 
- [x] Tube 
- [x] Torus 
- [x] TorusKnot 
- [x] Ring 
- [x] Tetrahedron 
- [x] Polyhedron 
- [x] Icosahedron 
- [x] Octahedron 
- [x] Dodecahedron 
- [x] Extrude 
- [x] Lathe 
- [x] Parametric
# Abstractions
- [x] Line
- [x] Text
- [x] Detailed
- [x] PositionalAudio 
- [x] StandardEffects  
# Shaders
- [x] MeshWobbleMaterial
- [x] Sky
- [x] Stars
- [x] softShadows() 
# Misc
- [x] HTML
- [x] Shadow
- [x] Stats 
- [x] draco() 
- [x] meshBounds() 
- [x] useCamera() 
- [x] useHelper()  
- [x] useAspect()

### CODE REVIEW

**Comment by gsimone** [comment]
A possible way to add docs would be to use https://www.npmjs.com/package/@storybook/addon-docs

**Comment by Fasani** [comment]
The following paths create errors:
`path=/story/shapes--extrude`
`path=/story/shapes--lathe`
`path=/story/shapes--parametric`

File: .storybook/Setup.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+export function Setup({ children }) {
+    return (
+        <Canvas colorManagement shadowMap camera={{ position: [-5, 5, 5] }} pixelRatio={window.devicePixelRatio}>
+            {children}
+            <ambientLight intensity={0.8} />
+            <pointLight intensity={1} color={'ffffff'} position={[0, 6, 0]} />
+        </Canvas>
+    )
+}
```
Comment (Fasani) [inline .storybook/Setup.tsx:12]: You should add newlines at the end of files, most editors have this built-in or as a plugin when you press save. This is the reason why:
https://thoughtbot.com/blog/no-newline-at-end-of-file

The above page also has solutions for common editors.

File: .storybook/Setup.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Canvas, useFrame } from 'react-three-fiber'
+
+export function Setup({ children }) {
+    return (
+        <Canvas colorManagement shadowMap camera={{ position: [-5, 5, 5] }} pixelRatio={window.devicePixelRatio}>
+            {children}
+            <ambientLight intensity={0.8} />
+            <pointLight intensity={1} color={'ffffff'} position={[0, 6, 0]} />
```
Comment (Fasani) [inline .storybook/Setup.tsx:9]: The colour is missing a hash (#) you should add a hash, alternatively pointLight has a default colour of white, in theory the colour could be removed altogether.

**Comment by gsimone** [comment]
Yes, those shapes are missing the arguments; I have some open issues on my fork

File: .storybook/Setup.tsx (hunk 1 above)
Comment (gsimone) [reply .storybook/Setup.tsx:12]: I thought we had prettier taking care of that, I may have skipped hooks by mistake

**Comment by Fasani** [comment]
Morning, I have two additional thoughts/concerns about this PR.

1) I think if this was to be accepted there should be a pre-commit or pull request message/check to ensure "any new components are requested to be added to the storybook" as part of the commit.
2) I also think before merge this branch should be complete in terms of all components should be here in the initial merge to match the state of what exists today.

**Comment by gsimone** [comment]
I think number 1 could be more easily handled during PR review.
As for number 2, agree, the PR is 100% WIP and looking for feedback

File: .storybook/stories/Detailed.stories.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            </Detailed>
+            <OrbitControls enablePan={false} enableRotate={false} zoomSpeed={0.5} />
+        </>
+    )
+}
+
+export const DetailedSt = () => <DetailedScene />
+DetailedSt.story = {
+    name: 'Default',
+}
```
Comment (Fasani) [inline .storybook/stories/Detailed.stories.js:39]: I was looking into prettier and I believe prettier won't fix these end lines. If you use vscode set `files.insertFinalNewline": true`

Comment (gsimone) [reply .storybook/stories/Detailed.stories.js]: Changed the setting, maybe we should consider adding an .editorconfig file with the rules 👍

File: .storybook/stories/TransformControls.stories.js [hunk 4]
```diff
@@ -1,17 +1,20 @@
-import React from 'react'
+import React, { useRef, useEffect, useState } from 'react'
```
Comment (Fasani) [inline .storybook/stories/TransformControls.stories.js:1]: useState not needed

File: .storybook/stories/TransformControls.stories.js [hunk 5]
```diff
@@ -22,5 +25,37 @@ TransformControlsStory.story = {
...
+    </>
+  )
+}
+
+export const TransformControlsLockSt = () => <TransformControlsLockScene />
+
+TransformControlsLockSt.story = {
+  name: 'lock orbit controls while transforming',
+  decorators: [(storyFn) => <Setup controls={false} >{storyFn()}</Setup>],
+}
```
Comment (Fasani) [inline .storybook/stories/TransformControls.stories.js]: No newline at end of file

**Comment by gsimone** [comment]
@drcmda I'm almost done with this, will wait for an opinion before the last leg. Do you think this is something worth keeping? ( so, finishing the pr, mantaining it in the future, etc )

**Comment by gsimone** [comment]
Since there are only 4 stories missing, I'll set this as ready for review

**Comment by drcmda** [comment]
@gsimone  nice! can we merge it like that?

**Comment by gsimone** [comment]
Yes, you can merge, there are 2-3 stories missing but I'll add them in a bulk with the new stuff. Also a small docs paragraph to explain how to use the storybook to contribute 👍

**Comment by gsimone** [comment]
Well, @emmelleppi just added the last missing stories so we are even ready-er to merge 😄

**Comment by drcmda** [comment]
@gsimone could you fix the conflicting files? you can merge as well - im on a trip and dont have internet often :-)

**Pull request merged**